### PR TITLE
Adding integration tests

### PR DESCRIPTION
**Pull request by NachoPal**
Integration tests for parachains

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ polkadot_argument_parsing
 **/chains/
 *.iml
 .env
```
Comment (chevdor) [inline .gitignore:10]: is this `.env` just a helper or required to run the tests?
If so, I would suggest the following and provide an example.

```suggestion
.env
!.env-example
```

File: parachains/integration-tests/statemine/xcm/2_ump.yml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                    { v1: [ *ap_ksm_fungible ] }, # assets
+                    0, # feeAssetItem
+                    { unlimited: true } # weightLimit
+                  ]
+                  events:
+                    - name: polkadotXcm.Attempted
+                      attribute:
+                        type: XcmV2TraitsOutcome
+                        isError: true
+                        value: Barrier
```
Comment (chevdor) [inline parachains/integration-tests/statemine/xcm/2_ump.yml:193]:
```suggestion
                        value: Barrier

```

File: parachains/integration-tests/statemine/xcm/3_hrmp-open-channels.yml [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+                      attributes:
+                        type: XcmV2TraitsOutcome
+                        isComplete: true
+                        value: 4,000,000
+                    - name: hrmp.OpenChannelAccepted
+                      chain: *relay_chain
+
+
+            
+
```
Comment (chevdor) [inline parachains/integration-tests/statemine/xcm/3_hrmp-open-channels.yml:405]:
```suggestion
                      chain: *relay_chain

```

File: parachains/integration-tests/statemine/xcm/4_hrmp.yml [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+                        }
+                      ]
+                    }
+                  ]
+                  events:
+                    - name: xcmpQueue.Fail
+                      chain: *assets_parachain
+                      attributes:
+                        type: XcmV2TraitsError
+                        value: FailedToTransactAsset
```
Comment (chevdor) [inline parachains/integration-tests/statemine/xcm/4_hrmp.yml:350]:
```suggestion
                        value: FailedToTransactAsset

```

File: parachains/integration-tests/statemine/xcm/1_dmp.yml [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                    ]
+          - name: Should keep the balance of the receiver
+            actions:
+              - asserts:
+                  equal:
+                    args:
+                      [
+                        $balance_ap_receiver_before,
+                        $balance_ap_receiver_after
+                      ]
```
Comment (chevdor) [inline parachains/integration-tests/statemine/xcm/1_dmp.yml:250]:
```suggestion
                      ]

```

File: parachains/integration-tests/statemint/xcm/2_ump.yml [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+                    { v1: [ *ap_ksm_fungible ] }, # assets
+                    0, # feeAssetItem
+                    { unlimited: true } # weightLimit
+                  ]
+                  events:
+                    - name: polkadotXcm.Attempted
+                      attribute:
+                        type: XcmV2TraitsOutcome
+                        isError: true
+                        value: Barrier
```
Comment (chevdor) [inline parachains/integration-tests/statemint/xcm/2_ump.yml:193]:
```suggestion
                        value: Barrier
                                 
```

File: parachains/integration-tests/statemint/xcm/3_hrmp-open-channels.yml [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+                        value: Ok
+                    - name: polkadotXcm.Sent
+                    - name: ump.ExecutedUpward
+                      chain: *relay_chain
+                      attributes:
+                        type: XcmV2TraitsOutcome
+                        isComplete: true
+                        value: 4,000,000
+                    - name: hrmp.OpenChannelAccepted
+                      chain: *relay_chain
```
Comment (chevdor) [inline parachains/integration-tests/statemint/xcm/3_hrmp-open-channels.yml:400]:
```suggestion
                      chain: *relay_chain

```

**Review by chevdor — APPROVED** [review]
LGTM beside the spacing issues

File: .gitignore (hunk 1 above)
Comment (NachoPal) [reply .gitignore:10]: I dind't add the `.env`, only `bin`

Comment (chevdor) [reply .gitignore:10]: The question still stands :)
If the `.env` is not mandatory, we can skip this change.

**Comment by gilescope** [comment]
Might be nice to mention in the readme how to run the tests.

**Comment by NachoPal** [comment]
> Might be nice to mention in the readme how to run the tests.

Ok, I still have to create a script to run them easily. I'll add a mention in the README then.

**Comment by bkchr** [comment]
Why isn't that done with zombienet? We don't want to have X different testing tools that no one maintains. If there is support missing for stuff in zombienet, then we need to add support. XCM aka sending transactions is a really "standard" feature that zombienet should support. So, please port these tests to normal zombienet tests. (I know that zombienet can just run normal polkadot-js scripts, if that isn't enough we need to make it easier to send xcm messages)

CC @pepoviola

**Comment by NachoPal** [comment]
The tool was created to fulfil Statemine/t testing needs when zombienet was still in its very early stages and was something we couldn't use at all. Since zombienet became stable, we integrated it to the tool and now it can be used alongside to deploy the network.

Regarding your comment about "no one maintains", I do the same way Javier mantains zombienet, and the rest of the Common Good Parachains team can do it also + the community. I strongly believe in the virtues of the tool and how it eases writing tests quickly. The tool has its own npm package, so it can also be considered to be integrated directly in zombienet.

The tests have been in its own repository until now, and we have been using them already for moths, but now we need to add them here (it is just a bunch of yaml files) in our efforts to improve our release process. We need a CI job to run the tests.

**Pull request merged**